### ISSUE TITLE
4k60fps Video horrible performance after update

### ISSUE THREAD

**Issue by methane4** [Eloston/ungoogled-chromium#1701]
**Describe the bug**
After updating, trying to watch a 4k60fps video is impossible, very laggy and stuttery

**Expected behavior**
The video to play without issue as normal.

**Screenshots**
https://a.tmp.ninja/hkMOEChb.mp4

**Environment (please complete the following information):**
 - OS/Platform and version: Linux - Fedora 34
 - ungoogled-chromium version: 95.0.4638.54 (Flatpak)

**Comment by methane4** [comment]
Im now on 96 (flatpak cc @rany2 ) however even though that the stutters are gone and videos play fine it seems that hardware decoding doesnt really work.. should i open an issue?

**Comment by PF4Public** [comment]
> it seems that hardware decoding doesnt really work..

based on what?

**Comment by methane4** [comment]
> > it seems that hardware decoding doesnt really work..
> 
> based on what?

@PF4Public my cpu usage when watching a video..

**Comment by rany2** [comment]
@methane4 not sure, but someone reported the same thing for the Brave flatpak... I'll check but it might be a general Chromium bug

**Comment by rany2** [comment]
Seems the bug is still open, might not be fixed yet... https://bugs.chromium.org/p/chromium/issues/detail?id=1262051

**Comment by methane4** [comment]
> @methane4 not sure, but someone reported the same thing for the Brave flatpak... I'll check but it might be a general Chromium bug

also, it should be documented on the wiki how to enable it properly (if you write a simple guide i can submit a PR)... it can be confusing. I would also like to double check if i have it enabled correctly.

**Comment by methane4** [comment]
> Seems the bug is still open, might not be fixed yet... https://bugs.chromium.org/p/chromium/issues/detail?id=1262051

i will keep an eye on it

**Comment by rany2** [comment]
@methane4 for flatpak, it's a bit different... you should use `features+=` and `features-=` for `--enable-features` and `--disable-features` respectively... 

My config that seems to work fine is the following: 

```
$ cat ~/.var/app/com.github.Eloston.UngoogledChromium/config/chromium-flags.conf
--ignore-gpu-blocklist
--enable-gpu-rasterization
--enable-zero-copy

features+=VaapiVideoEncoder
features+=VaapiVideoDecoder

features+=Vulkan

--enable-oop-rasterization
features+=Vulkan
features+=CanvasOopRasterization
features+=RawDraw
```

**Comment by methane4** [comment]
> @methane4 for flatpak, it's a bit different... you should use `features+=` and `features-=` for `--enable-features` and `--disable-features` respectively...
> 
> My config that seems to work fine is the following:
> 
> ```
> $ cat ~/.var/app/com.github.Eloston.UngoogledChromium/config/chromium-flags.conf
> --ignore-gpu-blocklist
> --enable-gpu-rasterization
> --enable-zero-copy
> 
> features+=VaapiVideoEncoder
> features+=VaapiVideoDecoder
> 
> features+=Vulkan
> 
> --enable-oop-rasterization
> features+=Vulkan
> features+=CanvasOopRasterization
> features+=RawDraw
> ```

@rany2 okay, i didnt know i had to use features like that, i will test it. again, this needs to be better documented..

**Comment by methane4** [comment]
> @methane4 for flatpak, it's a bit different... you should use `features+=` and `features-=` for `--enable-features` and `--disable-features` respectively...
> 
> My config that seems to work fine is the following:
> 
> ```
> $ cat ~/.var/app/com.github.Eloston.UngoogledChromium/config/chromium-flags.conf
> --ignore-gpu-blocklist
> --enable-gpu-rasterization
> --enable-zero-copy
> 
> features+=VaapiVideoEncoder
> features+=VaapiVideoDecoder
> 
> features+=Vulkan
> 
> --enable-oop-rasterization
> features+=Vulkan
> features+=CanvasOopRasterization
> features+=RawDraw
> ```
![Screenshot from 2021-11-21 00-01-36](https://user-images.githubusercontent.com/46012160/142744423-f4fcc43c-06f8-4689-9885-5d4fabc83f4f.png)

@rany2 no luck, this was the result, videos dont decode at all.(audio did)

**Comment by rany2** [comment]
@methane4 try just having: 

```
features+=VaapiVideoEncoder
features+=VaapiVideoDecoder
```

in the config, I believe some people have this bug because of enabling vulkan...

**Comment by methane4** [comment]
> @methane4 try just having:
> 
> ```
> features+=VaapiVideoEncoder
> features+=VaapiVideoDecoder
> ```
> 
> in the config, I believe some people have this bug because of enabling vulkan...

@rany2 videos play, however it doesnt work, and also i drop a lot of frames and have a stutter from time to time.

**Comment by PF4Public** [comment]
> > features+=Vulkan

> @rany2 no luck, this was the result, videos dont decode at all.(audio did)

VAAPI doesn't work properly with Vulkan. That's a Chrome issue.

**Comment by methane4** [comment]
> @methane4 try just having:
> 
> ```
> features+=VaapiVideoEncoder
> features+=VaapiVideoDecoder
> ```
> 
> in the config, I believe some people have this bug because of enabling vulkan...

@rany2 i have an update, it seems that sometimes i have about 25% cpu usage, other times i have 50% or very high cpu usage, no matter what resolution. Strange

**Comment by methane4** [comment]
I think decoders are at blame, i see on the media tab of the dev tools that some decoders are hardware acelarated and other arent. Testing the same video multiple times shows that sometimes a hardware decoder is picked and other times not, so the issue feels even strange. 

These are the ones that i saw testing on youtube videos:
![Screenshot from 2021-11-22 19-52-49](https://user-images.githubusercontent.com/46012160/142926126-4aa6c78e-0265-4107-adce-7dd2855a9633.png)

![Screenshot from 2021-11-22 19-52-22](https://user-images.githubusercontent.com/46012160/142926143-c18177c1-17bc-43d8-9618-25c0513f6307.png)

I think when VDA gets selected it works alright? i will need to some more testing but it feels like it.

So, a possible solution should be forcing VDA, what can be done about this? 
also i should say that with VDA isnt perfect, 25% cpu usage is to laugh of but its a case of less bad at this point.

**Comment by rany2** [comment]
I think https://github.com/erkserkserks/h264ify is worth a shot then.

**Comment by methane4** [comment]
> I think https://github.com/erkserkserks/h264ify is worth a shot then.
> 
> Edit: https://github.com/alextrv/enhanced-h264ify seems more fine grained, though I'm not sure if they're still working considering the fact they seem unmaintained.

@rany2 i could try however the problem isnt really codecs at fault, due to the same video and the same codec having different outcomes in different ocasions, and i also would prefer to keep the ability to watch some videos in a better quality sometimes.

**Comment by rany2** [comment]
@methane4 the problem seems to be codecs, dav1d is a decoder for AV1 media and VDA is for VP8, VP9 and H264.

**Comment by methane4** [comment]
> @methane4 the problem seems to be codecs, dav1d is a decoder for AV1 media and VDA is for VP8, VP9 and H264.

@rany2 
oh alright understood, so AV1 decoding is not working, should we open a separate issue? It would also be good if anybody not on flatpak could test av1 decoding.

**Comment by rany2** [comment]
@methane4 Nope, this is to be expected. Not a lot of hardware has AV1 hardware decoding support.

**Comment by methane4** [comment]
> @methane4 Nope, this is to be expected. Not a lot of hardware has AV1 hardware decoding support.

Oh, this makes sense. i had no idea that only the fairly new intel's tiger lake and newer hardware had av1 support. No idea why youtube is offering av1 to my browser.

![Screenshot from 2021-11-23 03-03-58](https://user-images.githubusercontent.com/46012160/142965526-e11fb114-87cb-4e36-baf3-c3d6b7b12f88.png)

I have tried the extension but i would prefer not to use it as its unmaintained, i will be trying this option in youtube's settings(seems to be working).

**Comment by methane4** [comment]
Just something more, 4k works fine but 4k 60fps drops frames.
